spring-cloud-skipper/src/main/java/org/springframework/cloud/skipper/domain/deployer/ConfigurationMetadataPropertyEntity.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.cloud.skipper.domain.deployer;
+
+import org.springframework.boot.configurationmetadata.ConfigurationMetadataProperty;
+
+/**
+ * This entity class is used to extend boot's {@link ConfigurationMetadataProperty} so that
+ * we can add our own constructor and not to explicitly import entity classes from boot.
+ *
+ * @author Janne Valkealahti
+ *
+ */
+@SuppressWarnings("serial")
+public class ConfigurationMetadataPropertyEntity extends ConfigurationMetadataProperty {
```

Comment:
@jvalkeal Do you think we could possibly miss some new fields that would be added to `ConfigurationMetadataProperty` this way?

Reply:
Well then we need to fix it here. I don't expect boot classes to change that much.